### PR TITLE
put all gemmlowp common code in one place

### PR DESCRIPTION
**Pull request by askhade**
**Description**: Moving all gemmlowp common code in 1 file to remove all duplicates

**Motivation and Context**
- This is first PR in series of 2... In the next PR I will add another layer similar to math_cpu.cc for choosing between different options like gemmlowp and mlas based on build options and input data types

### CODE REVIEW

File: onnxruntime/core/util/gemmlowp_common_wrapper.h [hunk 1]
```diff
@@ -14,6 +14,10 @@
 #pragma warning(push)
 #pragma warning(disable : 4100)  //'identifier' : unreferenced formal parameter
 #endif
+#ifdef _MSC_VER
```
Comment (snnn) [inline onnxruntime/core/util/gemmlowp_common_wrapper.h]: could  it be moved up to the previous ifdef section?

Comment (askhade) [reply onnxruntime/core/util/gemmlowp_common_wrapper.h]: yes it can be, any specific reason?

Comment (snnn) [reply onnxruntime/core/util/gemmlowp_common_wrapper.h]: Just wondering why they are listed separately, not with the warning 4100.

Comment (askhade) [reply onnxruntime/core/util/gemmlowp_common_wrapper.h]: my bad I did not notice the if else in between ... I thought all lines from 8 till 15 were in ifdef __GNUC__
Will make the change

**Comment by askhade** [comment]
Windows nGraph CI pipeline succeeded on 8/9 around 5pm... for some reason this has not synced with git yet... merging to master now

**Pull request merged**